src/molecules/Select/ComboBox/ComboBox.tsx

```diff
@@ -21,9 +23,9 @@ export type ComboBoxProps<T extends SelectOptionRequired> = CommonSelectProps &
  * @param syncParentChildSelection - Syncs parents/child selections, default is true
  */
 export function ComboBox<T extends SelectOptionRequired>(
-  props: ComboBoxProps<T>
+  props: ListComboBoxProps<T> | GroupedComboBoxProps<T>
 ) {
-  if (props.groups && props.onAddItem) {
+  if ('groups' in props && props.onAddItem) {
```

Comment:
The other two places you added thie `'groups' in props` change you kept the props.groups check aswell. Just wanted to check if not keeping it here was intentional :D

Reply:
No I missed this one, pushed a fix for it now 😄